src/charm.py

```diff
@@ -626,19 +631,24 @@ def _provide_routed_ingress(self, relation: Relation):
         self._push_configurations(relation, config)
 
     def _provide_ingress(
-        self, relation: Relation, provider: Union[IngressPerAppProvider, IngressPerAppProvider]
+            self,
+            relation: Relation,
+            provider: Union[IngressPerAppProvider, IngressPerAppProvider, IngressPerLeaderProvider],
```

Comment:
Is this correct? I mean, the duplicated  `IngressPerAppProvider`

Reply:
fixing it